examples/nucleo_g474re/flash_json/main.cpp

```diff
@@ -0,0 +1,194 @@
+
+void
+to_json(json& j, const person& p)
+{
+	j = json{{"name", p.name}, {"height", p.height}, {"age", p.age}};
+}
+
+void
+from_json(const json& j, person& p)
+{
+	j.at("name").get_to(p.name);
+	j.at("height").get_to(p.height);
+	j.at("age").get_to(p.age);
+}
+}  // namespace data
+
+int
+main()
+{
+	Board::initialize();
+
+	MODM_LOG_INFO << "\n\n+++++++++++++++++++++++++++" << modm::endl;
+	MODM_LOG_INFO << "++ NLOHMANN JSON example ++" << modm::endl;
+	MODM_LOG_INFO << "+++++++++++++++++++++++++++\n" << modm::endl;
+
+	data::person alice_struct = {"Alice", 1.85, 80};
+	json alice_json = alice_struct;
+
+	// basic JSON character string usage
+	{
+		auto alice_json_string = alice_json.dump();
+
+		// imagine that we transmit this string over some interface
+		MODM_LOG_INFO << "JSON from struct is: " << alice_json_string.c_str() << modm::endl;
+
+		// and now we can retrieve the struct from the json object
+		json retrieved_alice_json = json::parse(alice_json_string);
+
+		// JSON is fairly self-annotated, so we might not even need the original struct
+		MODM_LOG_INFO << "\nRetrieved from string JSON has fields ... ";
+		for (auto& [key, value] : retrieved_alice_json.items())
+		{
+			MODM_LOG_INFO << key.c_str() << ", ";
+		}
+		MODM_LOG_INFO << modm::endl;
+		MODM_LOG_INFO.flush();
+
+		// check if retrieved struct is same
+		auto retrieved_alice_struct = retrieved_alice_json.template get<data::person>();
+		if (retrieved_alice_struct == alice_struct)
+		{
+			MODM_LOG_INFO << "\nRetrieved from string and original struct are identical"
+						  << modm::endl;
+		}
+		MODM_LOG_INFO.flush();
+	}
+
+	// basic binary JSON usage
+	{
+		// convert to BSON
+		auto alice_binary = json::to_bson(alice_json);
+
+		MODM_LOG_INFO << "\nBinary JSON (BSON): ";
+		for (const auto& byte : alice_binary) { MODM_LOG_INFO.printf(" 0x%02x", byte); }
+		MODM_LOG_INFO << modm::endl;
+
+		// put BSON in reserved flash
+		uint32_t err{0};
+		const size_t page_start =
+			Flash::getPage(reinterpret_cast<uint32_t>(&__flash_reserved_start));
+		const size_t num_bytes = alice_binary.size();
+		const size_t flash_page_size = Flash::getSize(page_start);
+		const size_t end_page = page_start + (num_bytes + flash_page_size - 1) / flash_page_size;
+		if (end_page >= max_flash_pages)
+		{
+			MODM_LOG_ERROR << "\nRequested flash end page exceeds flash [" << page_start << ", "
+						   << end_page << ")" << modm::endl;
+			MODM_LOG_INFO.flush();
+			while (1)
+				;
+		}
+
+		// erase the pages before programming
+		MODM_LOG_INFO << "\nErasing flash sectors [" << page_start << ", " << end_page << ")"
+					  << modm::endl;
+		MODM_LOG_INFO.flush();
+		if (not Flash::unlock())
+		{
+			MODM_LOG_INFO << "Flash unlock failed!" << modm::endl;
+			while (1)
+				;
+		}
+		for (size_t page{page_start}; page < end_page; page++) err |= Flash::erase(page);
+		if (err != 0)
+		{
+			MODM_LOG_ERROR << "\nThere was an error while erasing flash!" << modm::endl;
+			MODM_LOG_INFO.flush();
```

Comment:
Usage of `MODM_LOG_ERROR` and `MODM_LOG_INFO` is inconsistent.
Also I don't see why it is necessary to flush the ostream.

Reply:
Resolved this